EasyClangComplete.py

```diff
@@ -39,51 +33,68 @@
 Tools = tools.Tools
 PosStatus = tools.PosStatus
 
-# unfortunately because of how sublime text initializes the plugins I cannot
-# move these inside of some class.
-settings = None
-completer = None
-
 log = logging.getLogger(__name__)
 
+handle_plugin_loaded_function = None
+
 
 def plugin_loaded():
     """called right after sublime api is ready to use. We need it to initialize
     all the different classes that encapsulate functionality. We can only
     properly init them after sublime api is available."""
-    global settings
-    global completer
-    settings = plugin_settings.Settings()
-    # init the loggers
-    if settings.verbose:
-        logging.basicConfig(level=logging.DEBUG)
-    else:
-        logging.basicConfig(level=logging.INFO)
-
-    # init everythin else
-    if settings.use_libclang:
-        log.info(" init completer based on libclang")
-        completer = lib_complete.Completer(settings.clang_binary)
-        if not completer.valid:
-            log.error(" cannot initialize completer with libclang.")
-            log.info(" falling back to using clang in a subprocess.")
-            completer = None
-    if not completer:
-        log.info(" init completer based on clang from cmd")
-        completer = bin_complete.Completer(settings.clang_binary)
-    # As the plugin have just loaded, we might have missed an activation event
-    # for the active view so completion will not work for it until re-activated.
-    # Force active view initialization in that case.
-    EasyClangComplete.on_activated_async(sublime.active_window().active_view())
+    handle_plugin_loaded_function()
 
 
 class EasyClangComplete(sublime_plugin.EventListener):
-
     """Base class for this plugin. Most of the functionality is delegated
 
+    Attributes:
+        settings (plugin_settings.Settings): class that encapsulates settings
+        completer (plugin.completion.base_completion.BaseCompleter):
+            This object handles auto completion. It can be one of the following:
+            - bin_complete.Completer
+            - lib_complete.Completer
     """
-    @staticmethod
-    def on_activated_async(view):
+    settings = None
+    completer = None
+
+    def __init__(self):
+        super().__init__()
+        global handle_plugin_loaded_function
+        handle_plugin_loaded_function = lambda: self.on_plugin_loaded()
+        # By default be verbose and limit on settings change if verbose flag is
+        # not set.
+        logging.basicConfig(level=logging.DEBUG)
+
+    def on_plugin_loaded(self):
+        self.settings = plugin_settings.Settings()
+        self.on_settings_changed()
+        self.settings.add_change_listener(lambda: self.on_settings_changed())
```

Comment:
I saw myself linter complaining about E731 [1] but I don't see other way to do it.

[1] http://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def